src/javascript/core.js

```diff
@@ -6,69 +6,31 @@
 const Send = require('./core/send');
 const User = require('./core/user');
 const Session = require('./core/session');
-
-/**
- * Shared 'internal' scope.
- * @type {Object}
- */
 const settings = require('./core/settings');
 const utils = require('./utils');
 
-/**
- * Default properties for sending a tracking request.
- * @type {Object}
- * @return {Object} - The default settings for the component.
- */
-const defaultConfig = function () {
-	return {
-		async: true,
-		callback: function () {},
-		system: {},
-		context: {},
-		user: {
-			passport_id: utils.getValueFromCookie(/USERID=([0-9]+):/) || utils.getValueFromCookie(/PID=([0-9]+)\_/),
-			ft_session: utils.getValueFromCookie(/FTSession=([^;]+)/)
-		}
-	};
-};
+let rootID;
 
 /**
- * Generate and store a new rootID.
- * @param {string} new_id - Optional rootID, if you want to use your own. Otherwise we'll create one for you.
+ * Generate and store a new rootID, used to mark a new root event that
+ * subsequent events will be linked to.
  * @return {string|*} The rootID.
  */
-function setRootID(new_id) {
-	settings.set('root_id', requestID(new_id));
-	return settings.get('root_id');
+function setRootID() {
```

Comment:
While this is technically available externally if something `require()`s in the core object separately, it's not documented and I think it shouldn't be - this is internal behaviour and there's no advantage to allowing it to be customised, and just adds complexity.  It can still be overridden by supplying it as part of the event config.